arangod/StorageEngine/VPackSortMigration.cpp

```diff
@@ -0,0 +1,294 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2024 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Business Source License 1.1 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Max Neunhoeffer
+////////////////////////////////////////////////////////////////////////////////
+
+#include "VPackSortMigration.h"
+#include <rocksdb/utilities/transaction_db.h>
+
+#include "Indexes/Index.h"
+#include "Logger/LogMacros.h"
+#include "RestServer/DatabaseFeature.h"
+#include "RocksDBEngine/RocksDBComparator.h"
+#include "RocksDBEngine/RocksDBEngine.h"
+#include "RocksDBEngine/RocksDBIndex.h"
+#include "StorageEngine/EngineSelectorFeature.h"
+#include "StorageEngine/PhysicalCollection.h"
+#include "Utils/SingleCollectionTransaction.h"
+#include "Transaction/StandaloneContext.h"
+#include "VocBase/LogicalCollection.h"
+
+#include "Cluster/ClusterFeature.h"
+#include "Cluster/ClusterInfo.h"
+
+#include "Network/NetworkFeature.h"
+#include "Network/Methods.h"
+
+namespace arangodb {
+
+// On dbservers, agents and single servers:
+Result analyzeVPackIndexSorting(TRI_vocbase_t& vocbase, VPackBuilder& result) {
+  // Just for the sake of completeness, restrict ourselves to the RocksDB
+  // storage engine:
+  auto& engineSelectorFeature =
+      vocbase.server().getFeature<EngineSelectorFeature>();
+  if (!engineSelectorFeature.isRocksDB()) {
+    {
+      VPackObjectBuilder guard(&result);
+      guard->add("error", VPackValue(false));
+      guard->add("errorCode", VPackValue(0));
+      guard->add(
+          "errorMessage",
+          VPackValue("VPack sorting migration is unnecessary for storage "
+                     "engines other than RocksDB"));
+    }
+    return {};
```

Comment:
Shouldn't this return an error `Result` here?

Reply:
My thinking was that the API call was successful, even though unneeded. Let's discuss, happy to change. Here, the discussion is mute, since we do not have another storage engine.